mater/lib/src/stores/filestore.rs

```diff
@@ -90,6 +108,86 @@ where
         }
     }
 
+    let index_offset = position;
+    let single_width_index =
+        SingleWidthIndex::new(Sha256::output_size() as u32, entries.len() as u64, entries);
+    let index = Index::MultihashIndexSorted(MultihashIndexSorted::from_single_width(
+        SHA_256_CODE,
+        single_width_index.into(),
+    ));
+    writer.write_index(&index).await?;
+
+    writer.get_inner_mut().rewind().await?;
+    let header = CarV2Header::new(
+        false,
+        car_v1_start.try_into().unwrap(),
+        (index_offset - car_v1_start).try_into().unwrap(),
+        index_offset.try_into().unwrap(),
+    );
+    writer.write_header(&header).await?;
+
+    let header_v1 = CarV1Header::new(vec![root.unwrap()]);
+    writer.write_v1_header(&header_v1).await?;
+
+    writer.finish().await?;
+
+    Ok(root.unwrap())
+}
+
+async fn balanced_import_unixfs<Src, Out>(
```

Comment:
Those methods almost look exactly the same, I struggle to find the difference between those two.
1. Add docs for this one
2. Can we change it's more clear that most of the logic is actually the same? Like extract some methods etc?